packages/ra-ui-materialui/src/button/CloneButton.tsx

```diff
@@ -61,12 +56,12 @@ const sanitizeRestProps = ({
 }: Omit<CloneButtonProps, 'label' | 'scrollToTop' | 'icon'>) => rest;
 
 interface Props {
-    record?: Partial<RaRecord>;
+    record?: any;
```

Comment:
We should probably prefer `Record<string, unknown>` instead of `any`. Same everywhere we accept a `RecordType`